### PR TITLE
[stable/external-dns] Add missing flags related to processing sources, kubefed related

### PR DESCRIPTION
**Pull request by dionysius**
#### Is this a new chart

No.

#### What this PR does / why we need it:

There are a lot of options in external-dns which are not reflected in the helm chart. Especially the `*fqdn*` flags are quite required for coorporation with kubefed multicluster dns. But since I needed to touch the parameters list, I also added the other options.

#### Special notes for your reviewer:

- only new values
- I kept the same option level because of the existing ones, but we could consider moving all these "source related" options into their own key group
  - But then we have to move some existing values too, which would break the current api (or find a way to be compatible)
- Maybe look into PR https://github.com/helm/charts/pull/17490 for the other part of kubefed compatibility

#### Checklist
- [x] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md
- [x] Title of the PR starts with chart name (e.g. `[stable/chart]`)

### CODE REVIEW

File: stable/external-dns/README.md [hunk 1]
```diff
@@ -58,7 +58,13 @@ The following table lists the configurable parameters of the external-dns chart
...
 | `sources`                          | K8s resources type to be observed for new DNS entries by ExternalDNS                                     | `[service, ingress]`                                     |
 | `provider`                         | DNS provider where the DNS records will be created (mandatory) (options: aws, azure, google, ...)        | `aws`                                                    |
+| `namespace`                        | Limit sources of endpoints to a specific namespace (default: all namespaces)                             | `""`                                                     |
+| `fqdnTemplates`                    | A list of templated string that's used to generate DNS names from sources that don't define a hostname themselves (optional)   | `[]`                               |
+| `combineFQDNAnnotation`            | Combine FQDN template and annotations instead of overwriting (optional)                                  | `false`                                                  |
```
Comment (miguelaeh) [inline stable/external-dns/README.md]: Please remove the `(optional)` in all the lines, all of the parameters are supposed to be optional unless they are explicitly marked as mandatory.

File: stable/external-dns/Chart.yaml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: external-dns
-version: 2.6.4
+version: 2.6.5
```
Comment (miguelaeh) [inline stable/external-dns/Chart.yaml]: As you are adding new features this should be a minor version not a patch.

**Review by miguelaeh — CHANGES_REQUESTED** [review]
Thank you very much for the PR!!!
Please check the comments

File: stable/external-dns/README.md (hunk 1 above)
Comment (dionysius) [reply stable/external-dns/README.md]: :+1: https://github.com/helm/charts/pull/17494/commits/8c9dd3bc8861990633e73bd229737583f98eaf8f

File: stable/external-dns/Chart.yaml (hunk 2 above)
Comment (dionysius) [reply stable/external-dns/Chart.yaml]: :+1: https://github.com/helm/charts/pull/17494/commits/8c9dd3bc8861990633e73bd229737583f98eaf8f

**Comment by zanhsieh** [comment]
@dionysius 
Could you fix the conflicting files please?
stable/external-dns/Chart.yaml
stable/external-dns/README.md

**Comment by dionysius** [comment]
@zanhsieh done

**Comment by zanhsieh** [comment]
@prydonius
@tompizmor
@sameersbn
@carrodher
@javsalgar
@juan131
@miguelaeh
May you guys LGTM this PR please? Thank you.

**Comment by juan131** [comment]
Hi @dionysius @zanhsieh 

My apologies. We totally lost the track of this issue until I came across it today checking old tickets. Sorry for that.

**Comment by juan131** [comment]
Could you please reopen the PR so we can follow with the review process?

**Comment by dionysius** [comment]
Sure, lets try /reopen

**Pull request closed without merging**